Radar site threshold

AdenKoperczak opened the pull request:
Another quality of life feature for looking at nationwide radar/placefiles.

## Settings
1. Setting to zero disables threshold, but maybe a checkbox or tooltip should be added to make that clear.
2. Units. Currently uses nautical miles for constancy with placefiles. Could be changed to following distance units settings like alert radius.

## Linting
1. Magic numbers for settings default/min/max. I think adding constants for each would generally decrease readability. I think we should add `NOLINTBEGIN(readability-magic-numbers)`/`NOLINTEND` + comment to avoid this in the future if you agree.
2. `[[nodiscard]]`. This should probably be fixed for all settings in one pull or commit.
I can make a separate pull request for these if you think that would be best.

AdenKoperczak committed:
Add radar site threshold setting

AdenKoperczak committed:
Added inverted threshold for radar sites

AdenKoperczak committed:
fix formatting for radarSiteThreshold_.SetMaximum

AdenKoperczak committed:
Add NM unit suffix to radar site threshold spin box.

dpaulat commented:
> Another quality of life feature for looking at nationwide radar/placefiles.
> 
> ## Settings
> 1. Setting to zero disables threshold, but maybe a checkbox or tooltip should be added to make that clear.

I agree a tooltip would be appropriate here. A checkbox and second setting could be used too, but I don't think it's necessary. I do think 0 to disable is fine, but with placefiles, >= 999 also disables (at least in the shader, although placefile text doesn't respect this).

> 2. Units. Currently uses nautical miles for constancy with placefiles. Could be changed to following distance units settings like alert radius.

I agree following the distance units option for consistency is best. Most users don't deal directly with placefile contents, so I don't think consistency there is necessary.

> 
> ## Linting
> 1. Magic numbers for settings default/min/max. I think adding constants for each would generally decrease readability. I think we should add `NOLINTBEGIN(readability-magic-numbers)`/`NOLINTEND` + comment to avoid this in the future if you agree.
> 2. `[[nodiscard]]`. This should probably be fixed for all settings in one pull or commit.
>    I can make a separate pull request for these if you think that would be best.

I agree with both linting suggestions, and that both can be taken care of in another PR.

AdenKoperczak committed:
Added tooltips and changed units for radar site threshold.

AdenKoperczak committed:
add const for set unit callbacks

AdenKoperczak commented:
I added tooltip to this and alert radius, and changed the units to match with alert radius.
Just a thought, but if we wanted some consistency with placefiles, we could add nautical miles as an additional distance unit.
I found that the 999NM threshold was about where I wanted it for my setup, but it felt like it could be a little close in depending your setup, so I figured giving more of a range would be good.
